content/articles/looking-at-payments.md

```diff
@@ -0,0 +1,112 @@
+Note that this change only applies to B2C customers.
+
+This article will focus on what it means for a UK based VAT registered company as it is our case.
+
+
+# Business side
+If the customer is a business and is VAT registered, just getting the VAT number and checking it should be enough. However, companies paying without giving their VAT number or non-VAT registered ones should be treated as B2C.
+
+As mentioned in the introduction, we need to charge the VAT of the customer's country which means we need to locate them.
+
+Here are some examples of pieces of informations we can use to identify a customer's location:
+
+- billing address
+- IP address geolocation
+- bank location
+- card location
+
+Note that the legislation requires to have two non-conflicting items of information and keep them for 10 (!) years. 
+Without the non-conflicting information you have to reject the transaction. You could possibly ask for the user to send some proof by mail but that's a terrible user experience.
+
+If we manage to get customers to pay us, we then need to separate sales by country and apply the right VAT. This gives us the amount of tax we need to pay for each country.
+
+The UK has put in place something called the VAT Mini One Stop Shop or VAT MOSS (https://www.gov.uk/guidance/register-and-use-the-vat-mini-one-stop-shop). This allows businesses to register in only one country and let it redistribute the money after filing a VAT return for other countries on a quarterly basis with the amounts we calculated.
+
+
+# Coding
+Now that we have an overview of what the VAT system looks like, let's have a look at the solutions.
+
+## Payment providers
+None of these actually help with the VAT mess but are still needed.
+There are a few alternatives here:
+
+- [Stripe](https://stripe.com)
+- [Braintree](https://www.braintreepayments.com/)
+- [Paymill](https://www.paymill.com/)
+- [FastSpring](http://www.fastspring.com/)
+
+Paymill is based in Europe. Stripe is extremely easy to setup and get started. I haven't used the other two so I can't comment on those but it seems you need to be approved for a merchant account for them which means some some wait.
+
+Here are the advertised rates (you might get a better one by talking with them):
+- Paymill: 2.95% + 0.28€
+- Braintree: 2.4% + 20p (first £30k free) for EU cards, 3.4% + 20p for non EU cards
+- Stripe: 1.9% + 20p for UK cards and 2.9% + 20p for foreign cards
+- FastSpring: 5.9% + $.95 per transaction
+
+Let's use a small example to see the difference: £10k sales with £5k from the UK, £3k from NA and £2k from Europe with let's say 100 transactions:
+
+```python
+paymill = (2.95 * 10000 / 100) + (0.20 * 100) # assuming £1=1€
+>> 315
+braintree = (2.4 * 7000 / 100) + (3.4 * 3000 / 100) + (0.20 * 100)
+>> 290
+stripe = (1.9 * 5000 / 100) + (2.9 * 5000 / 100) + (0.20 * 100)
+>> 260
+fastspring = (5.95 * 10000 / 100) + (0.63 * 100) # today's rate $0.95 > £0.63
```

Comment:
fastspring doesn't compete with stripe, it competes with recurly etc but interesting nevertheless